### PR TITLE
Preparing OpenT2T package for publish

### PR DESCRIPTION
**Pull request by shalinijoshi19**
(1) Updating Package.json to rev'd up version; The previous publish before me, ended up publishing version 1.0.2 on npm but the corresponding change to package,json wasn't committed to GitHub. So now we need to jump from github's 1.0.1 to 1.0.3

(2) This isn't a breaking change but an additive one; Ideally i'd rev the middle version for a new feature (Error Handling) but this isn't part of our currently defined versioning story yet; So continuing with tradition of rev'ving minor.

(3) Thanks to Jason, there is a prepublish/postpublish script - tspublish.js - which is run as part of npm publish today. Updated the script to include readme.md to be included in the root node folder so its included in the npm package also. 

(4) Updating readme.md with publish workflow also.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+# Running the linter
+- npm run lint
+
+# Preparing the package for publishing
+1. Update version number in package.json; Send a PR out for approval/FYI
+2. Follow steps to have the changes merged to Master
+3. cd to the node folder and run the following:
+4. npm run build
+5. npm run prepublish
+6. npm publish
```
Comment (jasongin) [inline README.md]: Doesn't the prepublish script get run automatically by `npm publish`? And that would also include build. So you can skip steps 4 and 5.

File: node/.gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@ node_modules/
 !typings/**/*.d.ts
 !test/**/*.js
 !tspublish.js
+README.md
```
Comment (jasongin) [inline node/.gitignore]: Specify it as `/README.md` to prevent excluding any other READMEs that might be added in the tree.

File: README.md (hunk 1 above)
Comment (shalinijoshi19) [reply README.md]: aah you are right -- I was scared to run npm publish for fear of actually publishing the package but just did and gladly it failed .. confirming it runs  prepublish:

c:\opent2t\sj\opent2t\node>npm publish

> opent2t@1.0.3 prepublish c:\opent2t\sj\opent2t\node
> git clean -dfqx build/* && tsc && node ./tspublish.js prepublish

File: node/.gitignore (hunk 2 above)
Comment (shalinijoshi19) [reply node/.gitignore]: Got it. Thanks

Comment (shalinijoshi19) [reply node/.gitignore]: Baah that syntax doesn't seem to work. Sticking for now with README.md then?

Comment (shalinijoshi19) [reply node/.gitignore]: never mind - my bad -- Fixed.

**Comment by danielrischar** [comment]
:shipit:

**Review by avneet723 — APPROVED** [review]

**Pull request merged**